### ISSUE TITLE
Amplify CLI push failing while cloudformation stacks still updating

### ISSUE THREAD

**Issue by tomhirschfeld** [aws-amplify/amplify-cli#6744]
> **Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the
> [Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

having a strange issue where I will run "amplify push", go through the dialogue, the cloudformation scripts start running, and then abruptly stop with the following error: 

```
UPDATE_COMPLETE                     functionrequestUserCampaignStateChange    AWS::CloudFormation::Stack Thu Feb 25 2021 21:14:33 GMT-0800 (Pacific Standard Time)
UPDATE_COMPLETE                     apipaypalBonusPayout                      AWS::CloudFormation::Stack Thu Feb 25 2021 21:14:33 GMT-0800 (Pacific Standard Time)
1 21:14:33 GMT-0800 (Pacific Standard Time)
21:14:33 GMT-0800 (Pacific Standard Time)
UPDATE_COMPLETE_CLEANUP_IN_PROGRESS amplify-wesponsored-web-hirschdev-81836   AWS::CloudFormation::Stack Thu Feb 25 2021 21:14:32 GMT-0800 (Pacific Standard Time)
⠧ Updating resources in the cloud. This may take a few minutes...

UPDATE_COMPLETE functionsharedv3                          AWS::CloudFormation::Stack Thu Feb 25 2021 21:14:45 GMT-0800 (Pacific Standard Time)
UPDATE_COMPLETE functionapprovePurchasesForReportingGroup AWS::CloudFormation::Stack Thu Feb 25 2021 21:14:44 GMT-0800 (Pacific Standard Time)
⠹ Updating resources in the cloud. This may take a few minutes...

UPDATE_COMPLETE amplify-wesponsored-web-hirschdev-81836 AWS::CloudFormation::Stack Thu Feb 25 2021 21:14:56 GMT-0800 (Pacific Standard Time)
UPDATE_COMPLETE authwesponsoredweb1966eb90              AWS::CloudFormation::Stack Thu Feb 25 2021 21:14:56 GMT-0800 (Pacific Standard Time)
UPDATE_COMPLETE apiwesponsoredweb                       AWS::CloudFormation::Stack Thu Feb 25 2021 21:14:55 GMT-0800 (Pacific Standard Time)
⠏ Updating resources in the cloud. This may take a few minutes...Error updating cloudformation stack
⠏ Updating resources in the cloud. This may take a few minutes...

Following resources failed

✖ An error occurred when pushing the resources to the cloud

Resource is not in the state stackUpdateComplete
An error occurred during the push operation: Resource is not in the state stackUpdateComplete
```

checking the cloudformation stacks in the console, they are often still running can finish correctly without error, but my cli shows this error

**Amplify CLI Version**
You can use `amplify -v` to check the amplify cli version on your system

4.44.1

**To Reproduce**
Steps to reproduce the behavior or terminal output if applicable

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

- OS: [e.g. Mac/Windows/Ubuntu]
- Node Version. You can use `node -v` to check the node version on your system

**Log output**
Include any relevant log output under `~/.amplify/logs/amplify-cli-<issue-date>.log`

**Additional context**

**Comment by ammarkarachi** [comment]
@tomhirschfeld Can you share the logs under ~/.amplify/logs/amplify-cli-<issue-date>.log?

**Comment by tomhirschfeld** [comment]
yes, give me a bit, I have a pressing client issue but I will try to get these in the next few days

**Comment by tomhirschfeld** [comment]
@ammarkarachi can you share a some kind of upload link so my logs aren't wide open for everyone on the internet to see?

**Comment by akshbhu** [comment]
@tomhirschfeld 

You can also send logs to amplify-cli@amazon.com with issue no. as subject and  redacting any secret information in it.

**Comment by tomhirschfeld** [comment]
sent

**Comment by akshbhu** [comment]
Hi @tomhirschfeld 

From the logs it seems you are getitng a throttling error while pushing stacks and other are validation as your pushing a stack which is already in ```UPDATE_IN_PROGRESS``` stage.

One thing you can try is to delete the stacks in AWS console which in ```UPDATE_IN_PROGRESS``` stage and do a ```amplify push ``` again

Let me know if this works for you?

**Comment by tomhirschfeld** [comment]
@akshbhu 

to walk through the experience one more time:

1) I run amplify push
2) amplify push runs for a few seconds and then fails with the error listed above in my earlier message
3) I check the console and the cloudformation stacks and they are still updating
4) I wait ~5 minutes
4) the cloudformation update finishes correctly and my changes are deployed

I am not blocked on deployment, but this is clearly a bug in amplify, it should not exit in this way when my stacks are still updating.

**Comment by akshbhu** [comment]
we have identified this a bug in catching throttling errors and deployment errors in CLI.

**Comment by ryan29871** [comment]
Any update on this as it still seems to be an issue with @aws-amplify/cli@5.5.0
The error appears when running the amplify push command from the cli.
When I look at the CloudFormation stacks amplify updates, it seems they continue to update and complete successfully after another 5-15 minutes...

The exact message below.

Following resources failed

× An error occurred when pushing the resources to the cloud

Resource is not in the state stackUpdateComplete
An error occurred during the push operation: Resource is not in the state stackUpdateComplete

**Comment by josefaidt** [comment]
Closing as we have since added better handling of rate limit/throttling exceptions. If you are still experiencing this issue please reply back to this thread or [create a new issue](https://github.com/aws-amplify/amplify-cli/issues/new?assignees=&labels=&template=1.bug_report.yaml) and we can investigate further 🙂
